stac_fastapi/pgstac/extensions/filter.py

```diff
@@ -40,3 +53,124 @@ async def get_queryables(
+        """
+        self.router.prefix = app.state.router_prefix
+        self.router.add_api_route(
+            name="Queryables",
+            path="/queryables",
+            methods=["GET"],
+            responses={
+                200: {
+                    "content": {
+                        "application/schema+json": {},
+                    },
+                    # TODO: add output model in stac-pydantic
+                },
+            },
+            response_class=self.response_class,
+            endpoint=create_async_endpoint(self.client.get_queryables, EmptyRequest),
+        )
+        app.include_router(self.router, tags=["Filter Extension"])
+
+
+@attr.s
+class ItemCollectionFilterExtension(FilterExtension):
+    """Item Collection Filter Extension."""
+
+    GET = FilterExtensionGetRequest
+    POST = FilterExtensionPostRequest
+
+    client: FiltersClient = attr.ib(factory=FiltersClient)
+    conformance_classes: List[str] = attr.ib(
+        default=[
+            FilterConformanceClasses.FILTER,
+            FilterConformanceClasses.FEATURES_FILTER,
+            FilterConformanceClasses.BASIC_CQL2,
+            FilterConformanceClasses.CQL2_JSON,
+            FilterConformanceClasses.CQL2_TEXT,
+        ]
+    )
+    router: APIRouter = attr.ib(factory=APIRouter)
+    response_class: Type[Response] = attr.ib(default=JSONSchemaResponse)
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        self.router.add_api_route(
+            name="Collection Queryables",
+            path="/collections/{collection_id}/queryables",
+            methods=["GET"],
+            responses={
+                200: {
+                    "content": {
+                        "application/schema+json": {},
+                    },
+                    # TODO: add output model in stac-pydantic
+                },
+            },
+            response_class=self.response_class,
+            endpoint=create_async_endpoint(self.client.get_queryables, CollectionUri),
+        )
+        app.include_router(self.router, tags=["Filter Extension"])
+
+
+@attr.s
+class CollectionSearchFilterExtension(FilterExtension):
+    """Collection Search Filter Extension."""
+
+    GET = FilterExtensionGetRequest
+    POST = FilterExtensionPostRequest
+
+    client: FiltersClient = attr.ib(factory=FiltersClient)
+    conformance_classes: List[str] = attr.ib(
+        default=[CollectionSearchConformanceClasses.FILTER]
+    )
+    router: APIRouter = attr.ib(factory=APIRouter)
+    response_class: Type[Response] = attr.ib(default=JSONSchemaResponse)
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        pass
```

Comment:
maybe there will be place for `/collections/queryables` in the future 🤷

Reply:
The pgstac queryables function actually supports this already

Reply:
🦸 I think we'll wait until there is a specification for this 🙏

Reply:
It is specified, see https://github.com/stac-api-extensions/collection-search/?tab=readme-ov-file#filter-cql

Reply:
It just doesn't pre-defined the location and lets people choose a place for it. It can be `GET /collections/queryables` if there's no conflict with a collection ID. That's why it's open, you can also do `GET /collection_queryables` or so. It's discovered via links. The link structure is the same as for other queryables, delivered via `GET /collections` in the links.

Reply:
> It can be GET /collections/queryables if there's no conflict with a collection ID.

😅 I didn't think about that

Reply:
Not sure who'd be so dumb to name a collection queryables ;-), but it could happen. And some servers may not like to resolve such conflicts in the router, so that's why the specification kept it open to be hosted at any path. So feel free to choose what path seems most reasonable to you.